core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java

```diff
@@ -0,0 +1,118 @@
+package ca.jrvs.apps.grep;
+
+import org.apache.log4j.BasicConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JavaGrepImp implements JavaGrep {
+
+    private static final Logger logger = LoggerFactory.getLogger(JavaGrep.class);
+    private String regex;
+    private String rootPath;
+    private String outFile;
+
+    public static void main(String[] args) {
+        if (args.length != 3) {
+            throw new IllegalArgumentException("Must provide: regex rootPath outFile");
+        }
+
+        BasicConfigurator.configure();
+        JavaGrepImp javaGrepImp = new JavaGrepImp();
+        javaGrepImp.setRegex(args[0]);
+        javaGrepImp.setRootPath(args[1]);
+        javaGrepImp.setOutFile(args[2]);
+
+        try {
+            javaGrepImp.process();
+        } catch (Exception e) {
+            logger.info(e.getMessage());
```

Comment:
logger.info("meaningful message", e);
            
            logger will print e.stacktrack so you don't need to print e.getMessage